src/models/PropStore.ts

```diff
@@ -0,0 +1,49 @@
+import { InjectionKey } from 'vue'
+import { CreateRouteOptions, isWithComponent, isWithComponents } from '@/types/createRouteOptions'
+import { ResolvedRoute } from '@/types/resolved'
+import { MaybePromise } from '@/types/utilities'
+
+export const propStoreKey: InjectionKey<PropStore> = Symbol()
+
+type NamedComponentProps = { name: string, props?: (params: Record<string, unknown>) => MaybePromise<Record<string, unknown>> }
+
+export class PropStore {
+  private readonly store = new Map<string, MaybePromise<Record<string, unknown>>>()
```

Comment:
I think if we make this a `reactive` then the props in router view will automatically be reactive without the hack of accessing `route.params`.

Reply:
Separately, I'm wondering if we even need this specific of a type. Could it just be `store = new Map<string, unknown>()`?